codegenerator/cli/templates/static/codegen/src/Index.res

```diff
@@ -1,46 +1,153 @@
+  S.schema(s => Active({
+    envioVersion: s.matches(S.option(S.string)),
+    chains: s.matches(S.array(chainDataSchema)),
+    indexerStartTime: s.matches(S.datetime(S.string)),
+    isPreRegisteringDynamicContracts: s.matches(S.bool),
+    isUnorderedMultichainMode: s.matches(S.bool),
+    rollbackOnReorg: s.matches(S.bool),
+  })),
+])
 
-      logger.udebug(args.length > 1 ? args : args[0]);
-    };
-    console.warn = function() {
-      var args = Array.from(arguments);
+// let setApiTokenEnv = {
+//   let initialted = ref(None)
+//   let envPath = NodeJs.Path.resolve([".env"])
+//   async apiToken => {
+//     // Execute once even with multiple calls
+//     if initialted.contents !== Some(apiToken) {
+//       initialted := Some(apiToken)
 
-      logger.uwarn(args.length > 1 ? args : args[0]);
-    };
-    console.error = function() {
-      var args = Array.from(arguments);
+//       let tokenLine = `ENVIO_API_TOKEN="${apiToken}"`
 
-      logger.uerror(args.length > 1 ? args : args[0]);
-    };
-  }
-`)
-// overrideConsoleLog(Logging.logger)
+//       try {
+//         // Check if file exists
+//         let exists = try {
+//           await NodeJs.Fs.Promises.access(envPath)
+//           true
+//         } catch {
+//         | _ => false
+//         }
 
-{
+//         if !exists {
+//           // Create new file if it doesn't exist
+//           await NodeJs.Fs.Promises.writeFile(
+//             ~filepath=envPath,
+//             ~content=tokenLine ++ "\n",
+//             ~options={encoding: "utf8"},
+//           )
+//         } else {
+//           // Read existing file
+//           let content = await NodeJs.Fs.Promises.readFile(~filepath=envPath, ~encoding=Utf8)
+
+//           // Check if token is already set
+//           if !Js.String.includes(content, "ENVIO_API_TOKEN=") {
+//             // Append token line if not present
+//             await NodeJs.Fs.Promises.appendFile(
+//               ~filepath=envPath,
+//               ~content="\n" ++ tokenLine ++ "\n",
+//               ~options={encoding: "utf8"},
+//             )
+//           }
+//         }
+//       } catch {
+//       | Js.Exn.Error(err) => {
+//           Js.Console.error("Error setting up ENVIO_API_TOKEN to the .env file:")
+//           Js.Console.error(err)
+//         }
+//       }
+//     }
+//   }
+// }
+
+let startServer = (~getState, ~shouldUseTui as _) => {
   open Express
 
   let app = makeCjs()
 
-  app->use(jsonMiddleware())
-
   app->get("/healthz", (_req, res) => {
     // this is the machine readable port used in kubernetes to check the health of this service.
     //   aditional health information could be added in the future (info about errors, back-offs, etc).
-    let _ = res->sendStatus(200)
+    res->sendStatus(200)
+  })
+
+  app->useFor("/console", (req, res, next) => {
+    switch req.headers->Js.Dict.get("origin") {
+    | Some(origin) if origin === "https://envio.dev" || origin === "http://localhost:3000" =>
```

Comment:
Can we not use different ports on localhost for this?

Reply:
Every port is a different origin. I made it use env var instead, to improve dev experience